drivers/sensor/bh1749/bh1749.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <sensor.h>
+#include <i2c.h>
+#include <misc/__assert.h>
+#include <misc/byteorder.h>
+#include <init.h>
+#include <kernel.h>
+#include <string.h>
+#include <logging/log.h>
+
+#include "bh1749.h"
+
+static struct k_delayed_work bh1749_init_work;
+static struct device *bh1749_dev;
+
+LOG_MODULE_REGISTER(BH1749, CONFIG_SENSOR_LOG_LEVEL);
+
+static const s32_t async_init_delay[ASYNC_INIT_STEP_COUNT] = {
+	[ASYNC_INIT_STEP_I2C] = 2,
+	[ASYNC_INIT_STEP_RESET_CHECK] = 0,
+	[ASYNC_INIT_RGB_ENABLE] = 0,
+	[ASYNC_INIT_STEP_CONFIGURE] = 0
+};
+
+
+static int bh1749_async_init_i2c(struct bh1749_data *data);
+static int bh1749_async_init_reset_check(struct bh1749_data *data);
+static int bh1749_async_init_rgb_enable(struct bh1749_data *data);
+static int bh1749_async_init_configure(struct bh1749_data *data);
+
+
+static int (* const async_init_fn[ASYNC_INIT_STEP_COUNT])
+			(struct bh1749_data *data) = {
+	[ASYNC_INIT_STEP_I2C]  = bh1749_async_init_i2c,
+	[ASYNC_INIT_RGB_ENABLE] = bh1749_async_init_rgb_enable,
+	[ASYNC_INIT_STEP_RESET_CHECK] = bh1749_async_init_reset_check,
```

Comment:
Swap the above two lines, please, to avoid confusion regarding the order of execution.

Reply:
@anangl Done